[Bug]: Data node panic close of a closed channel

### Problem encountered

The CI is failing for the data node integration steps with the data node panicking when closing a closed channel

### Observed behaviour

Context cancelled
panic: close of closed channel

### Expected behaviour

The data node integration tests pass

### Steps to reproduce

see:
https://jenkins.vega.rocks/blue/organizations/jenkins/vega/detail/PR-9421/17/pipeline/212


### Software version

develop

### Failing test

?

### Jenkins run

https://jenkins.vega.rocks/blue/organizations/jenkins/vega/detail/PR-9421/17/pipeline/212

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -218,6 +218,7 @@
 - [9412](https://github.com/vegaprotocol/vega/issues/9412) - Use vote close time for opening auction start time.
 - [9487](https://github.com/vegaprotocol/vega/issues/9487) - Reset auction trigger appropriately when market is resumed via governance.
 - [9489](https://github.com/vegaprotocol/vega/issues/9489) - A referrer cannot join another team.
+- [9441](https://github.com/vegaprotocol/vega/issues/9441) - fix occasional double close of channel in integration tests
 - [9074](https://github.com/vegaprotocol/vega/issues/9074) - Fix error response for `CSV` exports.
 - [9512](https://github.com/vegaprotocol/vega/issues/9512) - Allow hysteresis period to be set to 0.
 - [9526](https://github.com/vegaprotocol/vega/issues/9526) - Rename go enum value `REJECTION_REASON_STOP_ORDER_NOT_CLOSING_THE_POSITION` to match `db` schema

--- a/datanode/integration/integration_test.go
+++ b/datanode/integration/integration_test.go
@@ -107,12 +107,9 @@ func TestMain(m *testing.M) {
 	wg := sync.WaitGroup{}
 	wg.Add(1)
 
-	stopper := make(chan struct{})
-
 	go func() {
 		defer wg.Done()
-		if err := runTestNode(ctx, cfg, vegaHome, stopper); err != nil {
-			close(stopper)
+		if err := runTestNode(ctx, cfg, vegaHome); err != nil {
 			cfunc()
 			log.Fatal("running test node: ", err)
 		}
@@ -120,13 +117,12 @@ func TestMain(m *testing.M) {
 
 	client = graphql.NewClient(fmt.Sprintf("http://localhost:%v/graphql", cfg.Gateway.Port))
 	if err = waitForEpoch(client, lastEpoch, playbackTimeout); err != nil {
-		close(stopper)
 		cfunc()
 		log.Fatal("problem piping event stream: ", err)
 	}
 	// normal run - services should be terminated properly
 	if blockWhenDone == nil || !*blockWhenDone {
-		go handleSignal(ctx, cfunc, stopper)
+		go handleSignal(ctx, cfunc)
 	}
 
 	// Cheesy sleep to give everything chance to percolate
@@ -151,25 +147,22 @@ func TestMain(m *testing.M) {
 		os.Exit(0)
 	}
 
-	// this will close the stopper channel
 	cfunc()
 	wg.Wait()
 }
 
-func handleSignal(ctx context.Context, cfunc func(), sCh chan struct{}) {
+func handleSignal(ctx context.Context, cfunc func()) {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, syscall.SIGTERM, syscall.SIGINT)
 	for {
 		select {
 		case sig := <-c:
 			log.Printf("Received %+v signal", sig)
-			close(sCh) // close stopper channel
-			cfunc()    // cancel context
+			cfunc() // cancel context
 			return
 		case <-ctx.Done():
 			// context was cancelled for some reason, close stopper channel
 			log.Printf("Context cancelled")
-			close(sCh)
 			return
 		}
 	}
@@ -266,7 +259,7 @@ func newTestConfig(postgresRuntimePath string) (*config.Config, error) {
 	return &cfg, nil
 }
 
-func runTestNode(ctx context.Context, cfg *config.Config, vegaHome string, stopper chan struct{}) error {
+func runTestNode(ctx context.Context, cfg *config.Config, vegaHome string) error {
 	vegaPaths := paths.New(vegaHome)
 
 	loader, err := config.InitialiseLoader(vegaPaths)
@@ -289,11 +282,6 @@ func runTestNode(ctx context.Context, cfg *config.Config, vegaHome string, stopp
 		Version:     "test",
 		VersionHash: "",
 	}
-
-	go func() {
-		<-stopper
-		cmd.Stop()
-	}()
 
 	if err = cmd.Run(ctx, configWatcher, vegaPaths, []string{}); err != nil {
 		return fmt.Errorf("couldn't run node: %w", err)

